tests/drivers/watchdog/wdt_basic_api/testcase.yaml

```diff
@@ -45,6 +45,8 @@ tests:
     filter: dt_compat_enabled("st,stm32-window-watchdog") or dt_compat_enabled("st,stm32-watchdog")
     extra_args: DTC_OVERLAY_FILE="boards/stm32_wwdg_h7.overlay"
     platform_allow:
+      - nucleo_h7s3l8
+      - stm32h7s78_dk
```

Comment:
Adding `stm32h7s78_dk` to this test suite is not necessarily part of introducing the `nucleo_h7s3l78` but it is very minor so I think its fine :)